docs/testbed/README.testbed.Testbed_v2.hld.md

```diff
@@ -0,0 +1,127 @@
+# Testbed V2 Design
+# High Level Design Document
+
+
+# Table of Contents
+  * [About this Manual](#about-this-manual)
+  * [Requirement Overview](#Requirement-Overview)
+  * [Components](#Components)
+  * [Database Schema](#Database-Schema)
+  * [Implementation Plan](#Implementation-Plan)
+
+
+## About this Manual
+This documentation provides general information about the Testbed V2 feature implementation for `sonic-mgmt`. Testbed V2 aims to support multi-DUTs deployment with dynamic VLAN assignment and inter-DUTs link state propagation.
+
+## Requirement Overview
+* Maintain compatibility with existing testbed.
+* Support testbed/topology with multiple DUTs that have inter-DUTs connections.
+* Support flexible VLAN assignment.
+
+## Components
+* `test user`
+* `connection_db`
+  * Redis databases running on certain test server hosting connection-related metadata.
+* `servercfgd`
+  * running on the same server as `connection_db`
+  * a RPC server that is responsible for `connection_db` initial setup and provision.
+* `labcfgd`
+  * running on root/fanout switches
+  * subscribe to keyspace events of `connection_db` and act accordingly.
+
+## Database Schema
+* `LAB_CONNECTION_GRAPH_VERSIONS`
+```
+; ZSET that stores md5sum values of connection graph files that are used to provision `connection_db`
+key                              = LAB_CONNECTION_GRAPH_VERSIONS
+```
+* `LAB_META`
+```
+; Defines database metadata
+key                              = LAB_META
+; field                          = value
+ServerState                      = "active"/"provisioning"/"down"
+```
+* `SWITCH_TABLE`
+```
+; Defines root/leaf fanout switch metadata
+key                              = SWITCH_TABLE:switch_name
+; field                          = value
+HwSku                            = "FanoutRoot"/"FanoutLeaf"/"FanoutLeafSonic"
+ManagementIp                     =
+Type                             = "leaf_fanout"/"root_fanout"
```

Comment:
Does this need to be a totally separate table from the DUT_TABLE or can we have a general table with switch info? It seems like there is a lot of overlap in the fields here.

Reply:
I believe it will be convenient to store DUTs in a separate table will make listing then more easier.

Reply:
Can't we just have a look-up table of DUTs that points to the device table?

Reply:
Yes, that is a viable solution, we could simply store the DUTs in a list or set.